src/requirements/requirement-frontend-utils.ts

```diff
@@ -147,6 +150,21 @@ export function getUserRequirements({
         );
       })
       .flat(),
+    ...grad
+      .map(gradProg => {
+        const gradRequirement = requirementJson.grad[gradProg];
+        if (gradRequirement == null) return [];
+        return gradRequirement.requirements.map(
+          it =>
+            ({
+              ...it,
+              id: `Grad-${gradProg}-${it.name}`,
+              sourceType: 'Grad',
+              sourceSpecificName: gradProg,
+            } as const)
+        );
+      })
+      .flat(),
```

Comment:
We have an existing test that asserts all the requirement ID is stable: https://github.com/cornell-dti/course-plan/blob/c458ca6892cf6b9b73700e4284453cf45f724c4c/src/requirements/data/requirement-data-consistency.test.ts#L3-L29
You should extend that array with all the grad IDs.

Reply:
Thanks, I added grad programs to this test!